content/blaze.md

```diff
@@ -0,0 +1,559 @@
+
+Also, note that Spacebars is very forgiving of `null` values. It will not complain if you try to access a property on a `null` value (for instance `foo.bar` if `foo` is not defined), but instead simply treat it also as null. However there are exceptions to this---trying to call a `null` function, or doing the same *within* a helper will lead to exceptions.
+
+<h3 id="helpers">Calling helpers with arguments</h3>
+You can provide arguments to a helper like `checkedClass` by simply placing the argument after the helper call, as in: `{{checkedClass todo true 'checked'}}`. You can also provide a list of named keyword arguments to a helper with `{{checkedClass todo noClass=true classname='checked'}}. In this case, you might access those arguments with:
+
+```js
+Template.todosItem.helpers({
+  checkedClass(todo, kws) {
+    const classname = kws.hash.classname || 'checked';
+    if (todo.checked) {
+      return classname;
+    } else if (kws.hash.noClass) {
+      return `no-${classname}`;
+    }
+  }
+});
+```
+
+You can also pass the output of a helper to a template inclusion or other helper. To do so, use brackets to show precedence:
+
+```html
+{{> todosItem (todoArgs todo)}}
+```
+
+Here the `todo` is passed as argument to the `todoArgs` helper, then the output is passed into the `todosItem` template.
+
+<h3 id="inclusion">Template inclusion</h3>
+You "include" a sub-component with the `{{>` syntax. By default, the sub-component will gain the data context of the caller, although it's usually a good idea to be explicit. You can provide a single object as argument (as we did with the object returned by the `todoArgs` helper above), or provide a list of keyword arguments, as the `listShowPage` template does:
+
+```html
+{{> listsShow todosReady=Template.subscriptionsReady
+  list=(getFullList listIdOnly) todos=listIdOnly.todos}}
+```
+
+In this case, the `listShow` component can expect a data context of the form:
+
+```js
+{
+  todosReady: ...,
+  list: ...,
+  todos: ...
+}
+```
+
+<h3 id="helpers-in-tags">Helpers in tags</h3>
+We saw above that using a helper (or data context lookup) in the form `checked={{todo.checked}}` will add the checked property to the HTML tag if `todo.checked` evaluates to true. Also, you can directly include an object in the attribute list of an HTML element to set multiple attributes at once:
+
+```html
+<a {{attributes}}>My Link</a>
+```
+
+```js
+Template.foo.helpers({
+  attributes() {
+    return {
+      class: 'A class',
+      style: {background: 'blue'}
+    };
+  }
+});
+```
+
+<h3 id="rendering-html">Rendering pure HTML</h3>
+Although by default a mustache tag won't render any HTML, for [XSS](https://en.wikipedia.org/wiki/Cross-site_scripting) security reasons, you can render pure HTML with the triple-mustache: `{{{`.
```

Comment:
"won't render any HTML" -> "will escape HTML tags to avoid XSS"

"you can render pure HTML" -> "you can render raw HTML"